spec/unit/facter/login_defs_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'spec_helper'
+
+describe "custom fact login_defs" do
+
+  before(:each) do
+    Facter.clear
+
+    Facter.stubs(:value).with(:operatingsystem).returns('Linux')
+  end
+
+  context 'with a well formed /etc/login.defs' do
+    let(:login_defs_content) { <<-EOM
+MAIL_DIR        /var/spool/mail
```

Comment:
Please add some whitespace lines, comment lines, and empty lines to validate these lines are ignored.